pkg/registry/checkid/error.go

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2021 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package checkid
+
+import "fmt"
+
+// DuplicateError is an error type for the duplicate name error
+type DuplicateError struct {
```

Comment:
Can we make this private? As I can see this is public to be useable in the tests. So let's try to use chain with server/client to get the error naturally.

Reply:
We have different `setid`, `checkid` packages for client/server side (https://github.com/networkservicemesh/sdk/pull/789#discussion_r603436796). So this error can't be private.